feature/set.go

```diff
@@ -22,6 +22,10 @@ const (
 	// SetInvoice identifies features that should be advertised on invoices
 	// generated by the daemon.
 	SetInvoice
+
+	// SetInvoiceAmp identifies the features that should be advertised on
```

Comment:
> AMP invoices need to signal:
> - AMPRequired in order to avoid being paid by older clients that don't
   support it
> - Can't advertised MPP optional, otherwise older clients will attempt
   to pay the invoice with regular MPP payment.

> Hence, the features advertised on AMP invoices are mutually exclusive
from those advertised on MPP. Create a new set to classify the two.

Updated the commit message to this^^

Reply:
commit message not updated?

Reply:
I kinda like @joostjager's idea of removing the payment hash from the invoice though. Wouldn't this also prevent older clients from paying it?

Reply:
What do we even set as the payment hash in the invoice now? :p

Reply:
yep sorry, pushed now.

> What do we even set as the payment hash in the invoice now? :p

We choose a random payment hash, since we need _some_ payment hash to be able to insert into the invoice index which is indexed by payment hash.